### DISCUSSION TITLE
A "print_stack_trace" platform independent function in fmt package.

### DISCUSSION THREAD

**Discussion by xzores** [odin-lang/Odin#2449] [Ideas/Requests]
Hey, my biggest time sink using Odin currently is when I get a panic in some part of my code when the code is called from 50+ places.

I really would like to see a platform-independent print_stack_trace function, so that I could print the stack trace before I panic.
My current thoughts are that this feature would only work in debug mode.
While in debug mode all function calls would place themself at some stack (data structure) in global or heap memory. This way the program could at any time see how it get to where it is.

Then the fmt package (or some other package) could have a function that would look at that and just print the stack trace. This would be a high-value thing for me.

**Comment by Kelimion** [comment]
Moved to Ideas/Requests under Discussions.

**Comment by Lperlind** [comment]
In the meantime you can use a debugger to be able to get your call stack

**Reply by Beefster09** [reply]
this. even on non-debug builds, gdb can get a stack trace

**Comment by JesseRMeyer** [comment]
In the interim, if you're on Windows, https://github.com/DaseinPhaos/pdb is an option.

**Reply by xzores** [reply]
I have not gotten this to work.
